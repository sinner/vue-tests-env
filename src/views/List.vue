<template>
  <div class="list">
    <ul>
      <StrongRedListItem
        :key="index"
        v-for="(movie, index) in marvelMovies"
        :movie="movie"
      />
    </ul>
  </div>
</template>

<script>
import StrongRedListItem from '@/components/StrongRedListItem.vue';

export default {
  components: {
    StrongRedListItem,
  },

  data() {
    return {
      marvelMovies: [
        'Iron Man',
        'Avengers',
        'Infitinity War',
      ],
    };
  },

};
</script>

<style scoped>
.list {
  text-align: left;
}
</style>
