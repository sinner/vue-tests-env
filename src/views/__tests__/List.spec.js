import { mount } from '@vue/test-utils';
import List from '@/views/List.vue';

describe('List.vue', () => {
  it('should list element using mount', async () => {
    const wrapper = mount(List);
    const movies = wrapper.vm.marvelMovies;

    wrapper.setData({
      marvelMovies: [...movies, 'Endgame'],
    });
    expect(wrapper.vm.marvelMovies.length).toBe(4);

    wrapper.vm.$nextTick(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
