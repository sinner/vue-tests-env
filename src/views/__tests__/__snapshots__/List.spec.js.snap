// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue should list element using mount 1`] = `
<div class="list">
  <ul>
    <li class="list-item">
      <div class="colored-list-item">
        <div><span class="strong">
        - Iron Man
      </span></div>
      </div>
    </li>
    <li class="list-item">
      <div class="colored-list-item">
        <div><span class="strong">
        - Avengers
      </span></div>
      </div>
    </li>
    <li class="list-item">
      <div class="colored-list-item">
        <div><span class="strong">
        - Infitinity War
      </span></div>
      </div>
    </li>
    <li class="list-item">
      <div class="colored-list-item">
        <div><span class="strong">
        - Endgame
      </span></div>
      </div>
    </li>
  </ul>
</div>
`;
